chain/client/src/client_actor.rs

```diff
@@ -1961,6 +1961,89 @@ impl Handler<StateSplitResponse> for ClientActor {
     }
 }
 
+impl Handler<PreloadAccountData> for ClientActor {
+    type Result = Result<(), Error>;
+
+    fn handle(&mut self, msg: PreloadAccountData, _: &mut Self::Context) -> Self::Result {
+        let account_id = &msg.account_id;
+        let head = self.client.chain.head()?;
+        let head_header = self.client.chain.get_block_header(&head.last_block_hash)?;
+        let prev_block_hash = *head_header.prev_hash();
+        let shard_id = self
+            .client
+            .runtime_adapter
+            .account_id_to_shard_id(account_id, head_header.epoch_id())?;
+        let shard_uid =
+            self.client.runtime_adapter.shard_id_to_uid(shard_id, head_header.epoch_id())?;
+        let shard_state_root =
+            *self.client.chain.get_chunk_extra(&prev_block_hash, &shard_uid)?.state_root();
+        let trie = self.client.chain.runtime_adapter.get_trie_for_shard(
+            shard_id,
+            &prev_block_hash,
+            shard_state_root,
+        )?;
+
+        if trie.storage.as_caching_storage().is_none() {
+            return Err(Error::Other(
+                "account preloading attempt without a caching storage".to_owned(),
+            ));
+        }
+
+        let res = preload_account_state(account_id.clone(), &trie);
+        if let Err(e) = &res {
+            info!(target: "store", "Preloading cache with trie nodes for account {account_id} failed: {e}");
+        }
+
+        res.map_err(|e| Error::Other(e.to_string()))
+    }
+}
+
+fn preload_account_state(
+    account_id: AccountId,
+    trie: &near_store::Trie,
+) -> Result<(), near_store::StorageError> {
+    info!(target: "store", "Preloading cache with trie nodes for account {account_id}...");
+
+    let prefix = near_primitives::trie_key::trie_key_parsers::get_raw_prefix_for_contract_data(
+        &account_id,
+        &[],
+    );
+
+    // less than 10MB per thread should make it fairly quick.
+    // TODO: limit number of threads to something sensible.
+    let sub_trie_size = 10_000_000;
+
+    let mut state_iter = trie.iter()?;
+    state_iter.seek(prefix)?;
+
+    let mut handles = vec![];
+    for sub_trie in state_iter.heavy_sub_tries(sub_trie_size)? {
+        let TrieIterator { trie, trail, key_nibbles } = sub_trie?;
+        let storage = trie
+            .storage
+            .as_caching_storage()
+            .expect("preload called without caching storage")
+            .clone();
+        let root = trie.get_root().clone();
+        let handle = std::thread::spawn(move || {
+            trace!(target: "store", "Preload subtrie at {root}");
+            let trie = Trie::new(Box::new(storage), root, None);
+            let n = TrieIterator { trie: &trie, trail, key_nibbles }.count();
+            trace!(target: "store", "Preload subtrie at {root} done");
+            n
+        });
+        handles.push(handle);
+    }
+
+    let mut n = 0;
+    for handle in handles {
+        n += handle.join().expect("thread failed");
```

Comment:
`std::thread::scope` is a thing now! And, yeah, given that this is IO bound, I don't tihnk we should re-use our existing rayon's CPU pool.